gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/datafetcher/QueryServiceEntityFetcher.java

```diff
@@ -740,6 +737,41 @@ private QueryRequest buildTimeSeriesQueryRequest(
                             entityIdAttribute, Operator.NEQ, QUERY_SERVICE_NULL))
                 .map(Filter.Builder::build)
                 .collect(Collectors.toList()));
+
+    // Convert the existing filter into query service filter.
+    Filter queryFilter = convertToQueryFilter(entitiesRequest.getFilter()).build();
+    if (!Filter.getDefaultInstance().equals(queryFilter)) {
+      filterBuilder.addChildFilter(queryFilter);
+    }
+
+    // Time range is a mandatory filter for query service, hence add it if it's not already present.
+    if (!hasTimeRangeFilter(queryFilter, context.getTimestampAttributeId())) {
+      filterBuilder.addChildFilter(Filter.newBuilder().setOperator(Operator.AND)
+          .addChildFilter(createTimeFilter(context.getTimestampAttributeId(), Operator.GE, entitiesRequest.getStartTimeMillis()))
+          .addChildFilter(createTimeFilter(context.getTimestampAttributeId(), Operator.LT, entitiesRequest.getEndTimeMillis())));
+    }
+
+    return filterBuilder;
+  }
+
+  private static Filter createTimeFilter(String columnName, Operator op, long value) {
```

Comment:
nit: Why not use TimeRangeFilterUtil here?

Reply:
That's gateway-service filter vs query service filter. Thought about adding another one but naming is hard :)